### ISSUE TITLE
Random outputs (predictions)

### ISSUE THREAD

**Issue by render2020** [photopea/UNN.js#2]
Hi,

Sadly your script doesn't seems to be consistence in output (prediction).
Just run it few times (press F5 refresh Chrome console and you will see different results).
Every aprox. 5+ time it's random, not stable, doesn't matter how many iterations while training.

Run it yourself.
`
<script type="text/javascript" src="/js/UNN.js"></script>
<script type="text/javascript" src="/js/UNN.util.js"></script>

<script>
var net = UNN.Create([
["inpt","line",2,1,1],
["full","sigm",2,1,1],
["full","sigm",1,1,1] ],
0.5);

var O = [];

var In = [[0,0],[0,1],[1,0],[1,1]], Ou = [[0],[1],[1],[0]];
var prm = { method:"sgd", batch_size:1 };
for(var i=0; i< 50000; i++){

UNN.Train(net,In,Ou,prm);

}

UNN.GetOutput(net, [0,0], O);  console.log(O[2][0]);  //should be aprox 0.0xxxx
UNN.GetOutput(net, [1,0], O);  console.log(O[2][0]);  // should be aprox 0.9xxxx
UNN.GetOutput(net, [0,1], O);  console.log(O[2][0]);  // should be aprox 0.9xxxx

</script>`

FIREFOX:
![image](https://user-images.githubusercontent.com/76882763/103464946-c0baa500-4d72-11eb-8ff5-07e87ab58994.png)

CHROME:
![image](https://user-images.githubusercontent.com/76882763/103464803-75ec5d80-4d71-11eb-96c1-558bc950b70a.png)
![image](https://user-images.githubusercontent.com/76882763/103464806-80a6f280-4d71-11eb-8460-4df2a4fc0db9.png)
![image](https://user-images.githubusercontent.com/76882763/103464821-9c11fd80-4d71-11eb-8c03-2876d3877eef.png)
![image](https://user-images.githubusercontent.com/76882763/103464830-ba77f900-4d71-11eb-845e-35c689f63105.png)

IE EDGE
![image](https://user-images.githubusercontent.com/76882763/103464959-d4660b80-4d72-11eb-883d-658db17c9b2e.png)
![image](https://user-images.githubusercontent.com/76882763/103464963-daf48300-4d72-11eb-9eef-383bcdb74451.png)
![image](https://user-images.githubusercontent.com/76882763/103464974-f2cc0700-4d72-11eb-81b8-f9ebfaf4f44d.png)

**Comment by render2020** [comment]
Same code, formatted way:


**<script type="text/javascript" src="/js/UNN.js"></script>
<script type="text/javascript" src="/js/UNN.util.js"></script>

<script>
var net = UNN.Create([
["inpt","line",2,1,1],
["full","sigm",2,1,1],
["full","sigm",1,1,1] ],
0.5);

var O = [];

var In = [[0,0],[0,1],[1,0],[1,1]], Ou = [[0],[1],[1],[0]];
var prm = { method:"sgd", batch_size:1 };
for(var i=0; i< 50000; i++){

UNN.Train(net,In,Ou,prm);

}

UNN.GetOutput(net, [0,0], O);  console.log(O[2][0]);  //should be aprox 0.0xxxx
UNN.GetOutput(net, [1,0], O);  console.log(O[2][0]);  // should be aprox 0.9xxxx
UNN.GetOutput(net, [0,1], O);  console.log(O[2][0]);  // should be aprox 0.9xxxx

</script>**

**Comment by photopea** [comment]
Hi, I am aware of this. The back-propagation method is not guaranteed to reach a certain error level all the time. It is not a problem of UNN.js, but the same thing happens in any implementation of back-propagation learning.

You could improve it by trying different initial weights, , different learning method ("adadelta" instead of "sgd") or a different learning rate. Even the order of the training inputs is important.

If training the same network is more stable in another program, it is probably you are using different parameters (or their default parameters are different than ours). But for some other cases, our parameters could work better.

**Comment by render2020** [comment]
Hi again 

Thank you for your extensive explanation, I see 
It's my lack of knowledge 
I will try again and report.
Have good day and stay safe from Covid.
👌

**Comment by TrevorBlythe** [comment]
> Hi again
> 
> Thank you for your extensive explanation, I see It's my lack of knowledge I will try again and report. Have good day and stay safe from Covid. ok_hand

get good kid LMAOOOOOOOPOOOOOOO
